### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

File: .github/labeler.yml
Comment (ajschmidt8) [inline .github/labeler.yml:26]:
```suggestion
gpuCI:
```

File: .github/labeler.yml
Comment (ajschmidt8) [inline .github/labeler.yml:22]: No `CMake` label exists in this repo

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

**Comment by mike-wendt** [comment]
@jolorunyomi can you rebase? this shouldn't include any other changes besides the two config files

File: .github/labeler.yml
Comment (mike-wendt) [reply .github/labeler.yml]: @ajschmidt8 labels are auto-generated if they do not exist; just without color - if the cuGraph team wants this label/path combo they can update the color after this is merged

I see @jolorunyomi removed this, @rlratzel @BradReesWork do you want a `CMake` label?

Comment (mike-wendt) [reply .github/labeler.yml]: For reference cuML/RMM/cuDF have one for their CMake changes

Comment (rlratzel) [reply .github/labeler.yml]: The `cmake` label would be nice since we're making an effort to monitor cmake file changes more.

**Review by rlratzel — COMMENTED** [review]
I think this is probably fine as-is, but as mentioned in a comment, having the cmake label would be nice.

Also, our recently-updated `.github/CODEOWNERS` has many more extensions for docs than what's listed here. I don't know if we want docs limited to `.md` files for labels, but if not then I'd refer to the `CODEOWNERS` file to see a more exhaustive list.

Let me know what you want to do and I'll re-review, or if you don't feel those changes are appropriate I'll approve this.

File: .github/labeler.yml
Comment (mike-wendt) [reply .github/labeler.yml]: @jolorunyomi can you add this back to match the other repos?

**Comment by mike-wendt** [comment]
> I think this is probably fine as-is, but as mentioned in a comment, having the cmake label would be nice.
> 
> Also, our recently-updated `.github/CODEOWNERS` has many more extensions for docs than what's listed here. I don't know if we want docs limited to `.md` files for labels, but if not then I'd refer to the `CODEOWNERS` file to see a more exhaustive list.
> 
> Let me know what you want to do and I'll re-review, or if you don't feel those changes are appropriate I'll approve this.

@jolorunyomi can you take a look at the updated `CODEOWNERS` and port over the new docs paths?

File: .github/labeler.yml
Comment (jolorunyomi) [reply .github/labeler.yml:22]: the cmake config has been added.

**Comment by mike-wendt** [comment]
@rlratzel are these the paths you would like for the `doc` label?

https://github.com/rapidsai/cugraph/blob/branch-0.18/.github/CODEOWNERS#L4-L13

**Review by ajschmidt8 — APPROVED** [review]

**Comment by rlratzel** [comment]
> @rlratzel are these the paths you would like for the `doc` label?
> 
> From https://github.com/rapidsai/cugraph/blob/branch-0.18/.github/CODEOWNERS#L4-L13
> 
> ```
> datasets/          @rapidsai/cugraph-doc-codeowners
> notebooks/         @rapidsai/cugraph-doc-codeowners
> docs/              @rapidsai/cugraph-doc-codeowners
> **/*.txt           @rapidsai/cugraph-doc-codeowners
> **/*.md            @rapidsai/cugraph-doc-codeowners
> **/*.rst           @rapidsai/cugraph-doc-codeowners
> **/*.ipynb         @rapidsai/cugraph-doc-codeowners
> **/*.pdf           @rapidsai/cugraph-doc-codeowners
> **/*.png           @rapidsai/cugraph-doc-codeowners
> ```

Yes that looks good, thanks.

File: .github/labeler.yml
Comment (rlratzel) [reply .github/labeler.yml:22]: Looks good, thanks.

**Comment by mike-wendt** [comment]
> > @rlratzel are these the paths you would like for the `doc` label?
> > From https://github.com/rapidsai/cugraph/blob/branch-0.18/.github/CODEOWNERS#L4-L13
> > ```
> > datasets/          @rapidsai/cugraph-doc-codeowners
> > notebooks/         @rapidsai/cugraph-doc-codeowners
> > docs/              @rapidsai/cugraph-doc-codeowners
> > **/*.txt           @rapidsai/cugraph-doc-codeowners
> > **/*.md            @rapidsai/cugraph-doc-codeowners
> > **/*.rst           @rapidsai/cugraph-doc-codeowners
> > **/*.ipynb         @rapidsai/cugraph-doc-codeowners
> > **/*.pdf           @rapidsai/cugraph-doc-codeowners
> > **/*.png           @rapidsai/cugraph-doc-codeowners
> > ```
> 
> Yes that looks good, thanks.

@jolorunyomi can you make these changes to the config file for the `doc` label?

**Comment by mike-wendt** [comment]
Dropped python-codeowners from review as they were added before the rebase and this PR contains no python code

**Review by rlratzel — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**
